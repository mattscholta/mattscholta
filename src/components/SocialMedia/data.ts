import { Github, LinkedIn, StackOverflow, Twitter } from '@src/icons';
import { Icon } from '@src/types/icon';

interface Data extends Partial<HTMLAnchorElement> {
  icon: React.FC<Icon>;
  label: string;
}

// const data: Partial<HTMLAnchorElement>[] = [
const data: Data[] = [
  {
    href: 'https://github.com/visormatt',
    icon: Github,
    label: 'github',
    title: 'Find me on Github'
  },
  {
    href: 'https://www.linkedin.com/in/matthewscholta',
    icon: LinkedIn,
    label: 'linkedin',
    title: 'Connect on LinkedIn'
  },
  {
    href: 'https://stackoverflow.com/story/visormatt',
    icon: StackOverflow,
    label: 'stack-overflow',
    title: 'As required on StackOverflow'
  },
  {
    href: 'https://twitter.com/visormatt',
    icon: Twitter,
    label: 'twitter',
    title: 'Occasionally on Twitter'
  }
];

export { data };
export type { Data };
