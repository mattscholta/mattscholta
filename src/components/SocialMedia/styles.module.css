.component {
  align-items: center;
  display: flex;
  margin: var(--grid-6x) 0 var(--grid-2xs);
}

.icon {
  flex: 1 0;
  text-align: center;
}
