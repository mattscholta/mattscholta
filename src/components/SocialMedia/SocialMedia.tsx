import * as React from 'react';
import { color as colors } from '@barguide/style-guide';
import classnames from 'classnames';

import { analytics } from '@src/utils/analytics';
import { Data, data } from './data';

import styles from './styles.module.css';

export interface SocialMediaProps {
  className?: string;
  color?: string;
  fill?: string;
}

/**
 * @name SocialMedia
 * @description Simple component that wraps up my social media accounts
 */
const SocialMedia: React.FC<SocialMediaProps> = (props) => {
  const { className, color = colors.white, fill = colors.primary } = props;

  // Styles
  const css = classnames(styles.component, className);

  // Handlers
  const onClick = (event: React.MouseEvent<HTMLAnchorElement>) => {
    const { currentTarget } = event;
    const { gaLabel } = currentTarget.dataset;

    analytics.event('footer', 'click', gaLabel);
  };

  // Markup
  const renderLink = (link: Data) => {
    const { href, icon, label, title } = link;
    const Icon = icon;

    return (
      <a
        className={styles.icon}
        data-ga-label={label}
        href={href}
        key={label}
        onClick={onClick}
        rel="noopener noreferrer"
        target="_blank"
        title={title}
      >
        <Icon color={color} fill={fill} />
      </a>
    );
  };

  return <div className={css}>{data.map(renderLink)}</div>;
};

export { SocialMedia };
