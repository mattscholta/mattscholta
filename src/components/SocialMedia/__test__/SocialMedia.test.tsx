import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';

import { SocialMedia, SocialMediaProps } from '../SocialMedia';

describe('SocialMedia', () => {
  let props: SocialMediaProps;
  let wrapper: RenderResult;

  const createWrapper = () => <SocialMedia {...props} />;

  beforeEach(() => {
    props = {};
    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
