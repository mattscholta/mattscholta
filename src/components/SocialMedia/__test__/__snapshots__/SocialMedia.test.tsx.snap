// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialMedia matches snapshot 1`] = `
NodeList [
  <div
    class="component"
  >
    <a
      class="icon"
      data-ga-label="github"
      href="https://github.com/visormatt"
      rel="noopener noreferrer"
      target="_blank"
      title="Find me on Github"
    >
      <svg
        class="Github"
        color="#fff"
        height="49"
        viewBox="0 0 50 49"
        width="50"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#f00"
            fill-rule="nonzero"
          >
            <path
              d="M25.0023606,0 C11.195748,0 0,11.1936779 0,25.0023606 C0,36.0490583 7.16393731,45.4210103 17.0981401,48.7270289 C18.3476785,48.9584707 18.8064222, 48.1846513 18.8064222,47.5242756 C18.8064222,46.9280743 18.7832366,44.9585402 18.7724718,42.8693519 C11.816791,44.3817983 10.3490595,39.9193978 10.3490595, 39.9193978 C9.21172281,37.0294781 7.5729976,36.2610409 7.5729976,36.2610409 C5.30453473,34.7092617 7.74399135,34.7411418 7.74399135,34.7411418 C10.254661, 34.9175179 11.5766544,37.317642 11.5766544,37.317642 C13.8066125,41.1399543 17.4256367,40.0349118 18.8523794,39.3960656 C19.0767829,37.7801119 19.7247378, 36.6771396 20.439765,36.0527844 C14.8863992,35.4205626 9.0485956,33.2767225 9.0485956,23.6965141 C9.0485956,20.9668233 10.0252891,18.7364514 11.6246817, 16.9855243 C11.3650857,16.3557867 10.5092886,13.8128229 11.8668884,10.3689329 C11.8668884,10.3689329 13.9664276,9.69696439 18.7443179,12.93177 C20.7386939, 12.3777997 22.8775656,12.0999865 25.0023606,12.0904639 C27.1271555,12.0999865 29.2676834,12.3777997 31.2657854,12.93177 C36.0378794,9.69696439 38.1345202, 10.3689329 38.1345202,10.3689329 C39.4954325,13.8128229 38.6392213,16.3557867 38.3796253,16.9855243 C39.9827442,18.7364514 40.9528131,20.9668233 40.9528131, 23.6965141 C40.9528131,33.2994942 35.1038308,35.4139384 29.536388,36.0329111 C30.433174,36.8088009 31.2322491,38.3303558 31.2322491,40.6629933 C31.2322491, 44.0083447 31.2032673,46.7007727 31.2032673,47.5242756 C31.2032673,48.1896197 31.6533162,48.9692355 32.9206578,48.7237165 C42.8494783,45.4139716 50.004307, 36.045332 50.004307,25.0023606 C50.004307,11.1936779 38.8102153,0 25.0023606, 0 Z M9.36419706,35.6163904 C9.30913122,35.740599 9.1137097,35.7778615 8.93567741,35.6925718 C8.75433285,35.6110079 8.65248187,35.4416703 8.71127392, 35.3170478 C8.76509762,35.1891128 8.96093316,35.1535064 9.14186365,35.2392103 C9.32362225,35.3207739 9.42712939,35.4917678 9.36419706,35.6163904 Z M10.5940354, 36.7137588 C10.4747952,36.8243044 10.2416971,36.7729648 10.0835381,36.5982449 C9.91999679,36.4239389 9.8893587,36.1908407 10.010255,36.0786388 C10.1332215, 35.9680932 10.3592812,36.0198468 10.5232365,36.1941529 C10.6867778,36.370529 10.718658,36.601971 10.5940354,36.7137588 Z M11.437751,38.1178041 C11.2845605, 38.2242096 11.0340732,38.1244285 10.8792265,37.9020953 C10.7260358,37.6797619 10.7260358,37.4131274 10.8825387,37.306308 C11.0377994,37.1994887 11.2845605, 37.2955434 11.4414773,37.5162207 C11.5942539,37.7422802 11.5942539,38.0089145 11.437751,38.1178041 Z M12.8646548,39.7438869 C12.7276113,39.8950078 12.4357212, 39.8544328 12.2220824,39.6482467 C12.0034753,39.4466145 11.9426131,39.1605207 12.0800706,39.0094002 C12.2187701,38.8578659 12.5123164,38.9005108 12.7276113, 39.1050409 C12.9445623,39.306259 13.0108069,39.5944229 12.8646548,39.7438869 Z M14.7087894,40.2928342 C14.6483413,40.4886698 14.3672158,40.5776859 14.0840202,40.4944662 C13.8012387,40.4087624 13.6161679,40.1793907 13.6733038, 39.9814849 C13.7320959,39.7844074 14.0144634,39.6916649 14.2997291,39.7806809 C14.5820966,39.8659709 14.7675815,40.0936866 14.7087894,40.2928342 Z M16.807492, 40.5256834 C16.8145304,40.7318695 16.5743938,40.9028633 16.2771213,40.9065898 C15.9781927,40.9132142 15.7364,40.7463608 15.7330877,40.5434867 C15.7330877, 40.3352303 15.9678419,40.1658926 16.2667706,40.1609242 C16.5640432,40.1551278 16.807492,40.3207392 16.807492,40.5256834 Z M18.8692321,40.4466524 C18.9048386, 40.6478705 18.6982383,40.8544705 18.4030359,40.9095364 C18.1128019,40.9625319 17.8440972,40.8383234 17.8072487,40.6387617 C17.7712283,40.4325754 17.9815548, 40.2259751 18.2713748,40.1725657 C18.5669912,40.1212261 18.8315555,40.2421222 18.8692321,40.4466524 Z"
            />
          </g>
        </g>
      </svg>
    </a>
    <a
      class="icon"
      data-ga-label="linkedin"
      href="https://www.linkedin.com/in/matthewscholta"
      rel="noopener noreferrer"
      target="_blank"
      title="Connect on LinkedIn"
    >
      <svg
        height="50"
        viewBox="0 0 50 50"
        width="50"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill-rule="nonzero"
          >
            <path
              d="M46.1478327,0 L3.67820462,0 C1.64875462,0 0,1.6086089 0, 3.5920292 L0, 46.2402381 C0,48.2253888 1.64875462,49.83642 3.67820462,49.83642 L46.1478327, 49.83642 C48.1807435,49.83642 49.8371123,48.2253888 49.8371123,46.2402381 L49.8371123,3.5920292 C49.8371123,1.6086089 48.1807435,0 46.1478327,0"
              fill="#f00"
            />
            <path
              d="M10.8247396,6.92307692 C13.1885088,6.92307692 15.1085954, 8.84350954 15.1085954,11.2076247 C15.1085954,13.5734703 13.1885088, 15.4945951 10.8247396, 15.4945951 C8.45612538, 15.4945951 6.53846154,13.5734703 6.53846154,11.2076247 C6.53846154,8.84350954 8.45612538,6.92307692 10.8247396, 6.92307692 L10.8247396, 6.92307692 Z M7.12750038,18.7457293 L14.5192104,18.7457293 L14.5192104, 42.5291228 L7.12750038, 42.5291228 L7.12750038, 18.7457293 Z"
              fill="#fff"
            />
            <path
              d="M19.2307692,18.6687305 L26.3217304,18.6687305 L26.3217304, 21.9181343 L26.4200188,21.9181343 C27.4063646,20.0485764 29.8178935,18.0769231 33.4137296,18.0769231 C40.8988831, 18.0769231 42.2808058,23.0027687 42.2808058,29.4064026 L42.2808058,42.452124 L34.8942869,42.452124 L34.8942869, 30.8862674 C34.8942869,28.1269217 34.8434119,24.5795376 31.0530754,24.5795376 C27.2063269,24.5795376 26.6172881, 27.5842585 26.6172881,30.6865756 L26.6172881, 42.452124 L19.2307692,42.452124 L19.2307692,18.6687305"
              fill="#fff"
            />
          </g>
        </g>
      </svg>
    </a>
    <a
      class="icon"
      data-ga-label="stack-overflow"
      href="https://stackoverflow.com/story/visormatt"
      rel="noopener noreferrer"
      target="_blank"
      title="As required on StackOverflow"
    >
      <svg
        color="#fff"
        height="41"
        viewBox="0 0 50 50"
        width="50"
      >
        <path
          d="M4.16666 32.3792V48.4375C4.16666 49.3 4.86666 50 5.72916 50H44.2708C44.6852 50 45.0827 49.8354 45.3757 49.5424C45.6687 49.2493 45.8333 48.8519 45.8333 48.4375V32.3792C45.8333 30.3125 42.7083 30.3104 42.7083 32.3792V46.875H7.29166V32.3792C7.29166 30.3125 4.16666 30.3125 4.16666 32.3792Z"
          fill="#f00"
        />
        <path
          d="M13.8438 41.9708H34.1292C36.1958 41.9708 36.1979 38.8458 34.1292 38.8458H13.8438C11.7771 38.8458 11.775 41.9708 13.8438 41.9708ZM13.8354 32.3312C34.9875 36.6625 33.7063 36.4417 34.0146 36.4417C35.8833 36.4417 36.1604 33.725 34.3271 33.3479L14.4646 29.2687C12.4396 28.8542 11.8083 31.9146 13.8354 32.3312V32.3312ZM16.3625 21.5125C35.3646 30.9375 34.4708 30.7229 35.1667 30.7229C36.8125 30.7229 37.3333 28.4958 35.8646 27.7625L17.7583 18.7167C15.9146 17.7917 14.5125 20.5875 16.3625 21.5125V21.5125ZM20.9833 13.375L36.85 25.5854C38.4625 26.8292 40.4167 24.3875 38.7563 23.1104L22.8896 10.9C21.2521 9.63541 19.3438 12.1146 20.9833 13.375V13.375ZM43.1188 19.3937L31.1979 3.74166C29.9479 2.09791 27.4583 3.98958 28.7125 5.63541L40.6333 21.2875C41.875 22.9208 44.3813 21.0521 43.1188 19.3937Z"
          fill="#f00"
        />
      </svg>
    </a>
    <a
      class="icon"
      data-ga-label="twitter"
      href="https://twitter.com/visormatt"
      rel="noopener noreferrer"
      target="_blank"
      title="Occasionally on Twitter"
    >
      <svg
        color="#fff"
        height="41"
        viewBox="0 0 50 41"
        width="50"
      >
        <g
          fill="none"
          fill-rule="evenodd"
          stroke="none"
          stroke-width="1"
        >
          <g
            fill="#f00"
            fill-rule="nonzero"
          >
            <path
              d="M49.9996169,4.97067009 C48.1599515,5.78653128 46.1829373, 6.33793714 44.1078956,6.5859326 C46.2259159,5.31632748 47.8523345, 3.30602797 48.6182675, 0.910750218 C46.6363155,2.08616825 44.4409339,2.93988716 42.1043629,3.40003215 C40.2330579,1.40637538 37.5671067,0.160545645 34.6167654,0.160545645 C28.9514588, 0.160545645 24.3584219,4.75339973 24.3584219,10.4183404 C24.3584219, 11.2223139 24.4493169,12.0052553 24.6241574,12.7560086 C16.0986732, 12.3282347 8.54011526, 8.24435655 3.48071512,2.03825172 C2.59773421,3.55329187 2.09186737,5.31541306 2.09186737,7.19531379 C2.09186737,10.7541219 3.9030024,13.8939347 6.65545945, 15.7334173 C4.97380878,15.680197 3.39219756,15.2187718 2.00901931,14.4504614 C2.00828777,14.4932571 2.00828777,14.5362356 2.00828777,14.5793971 C2.00828777, 19.549548 5.54423492,23.6956081 10.2369456,24.6378444 C9.37609407,24.8723063 8.46988644,24.9975842 7.53423394,24.9975842 C6.87327846,24.9975842 6.23061175, 24.9333906 5.6044049,24.8135993 C6.90967309,28.8890648 10.698005,31.8547686 15.1867961,31.9376166 C11.6760874,34.6887934 7.25295294,36.3287458 2.4470349, 36.3287458 C1.61910314,36.3287458 0.802510401,36.2802806 0,36.1855448 C4.53963383,39.0960166 9.93170633,40.7941272 15.7246683,40.7941272 C34.592807, 40.7941272 44.9105889,25.1632802 44.9105889,11.6078408 C44.9105889,11.1630584 44.900713,10.7206535 44.8809612,10.280809 C46.8850425,8.83453455 48.6243028, 7.02778882 49.9996169,4.97067009"
            />
          </g>
        </g>
      </svg>
    </a>
  </div>,
]
`;
